### PR TITLE
Unbanish non-flaky tests, and generally tidy up e2e.

### PR DESCRIPTION
**Pull request by ghost**
I've checked all of the removed or amended regexes, and they're all either:
1. no longer flaky
2. redundant
3. insufficiently specific

It's theoretically possible to split this PR into multiple smaller ones, but sequencing that will be difficult, so I'd prefer to keep them all together if that's OK with the reviewer.  Let me know if the reason for any of the changes is not clear - they pretty much all fall into one of the above categories.

### CODE REVIEW

**Comment by quinton-hoole** [comment]
@ixdy for review too please.

**Comment by quinton-hoole** [comment]
The failures all look unrelated to this PR.  Retrying...
@k8s-bot test this again please

**Comment by ihmccreery** [comment]
LGTM once tests pass.

**Comment by quinton-hoole** [comment]
Assume flake.
@k8s-bot test this again please

File: hack/jenkins/e2e.sh [hunk 1]
```diff
@@ -116,40 +117,24 @@ GCE_SLOW_TESTS=(
...
 GCE_PARALLEL_FLAKY_TESTS=(
-    "DaemonRestart"
-    "Elasticsearch"
-    "PD"
-    "ServiceAccounts"
-    "Services.*change\sthe\stype"
-    "Services.*functioning\sexternal\sload\sbalancer"
-    "Services.*identically\snamed"
     "Services.*release.*load\sbalancer"
-    "Services.*endpoint"
```
Comment (ixdy) [inline hack/jenkins/e2e.sh]: on most recent PR Jenkins run, "Service endpoints latency should not be very high" failed. Seems it's still flaky?

Comment (quinton-hoole) [inline hack/jenkins/e2e.sh]: It's possible that it's been broken in the past few hours, after this PR was submitted, but it had a very clean bill of health before that:
job/kubernetes-e2e-gce-parallel/7717/testReport/(root)/Kubernetes%20e2e%20suite/Service_endpoints_latency_should_not_be_very_high/history/

But besides that, the regex that is being removed never matched that test anyway :-)

**Comment by quinton-hoole** [comment]
@k8s-bot test this again please

**Comment by quinton-hoole** [comment]
Looking a bit closer at the e2e errors, they actually look legitimate.  I'll have to debug this...

**Comment by thockin** [comment]
removing LGTM for now

**Comment by quinton-hoole** [comment]
Closing as obsolete.

**Pull request closed without merging**